PowerPointLabs/PowerPointLabs/ELearningLab/Views/ELearningWorkspaceViews/ELearningLabMainPanel.xaml.cs

```diff
@@ -536,6 +592,18 @@ private bool IsAzureVoiceSelected()
             }
             return false;
         }
+        private bool IsSlideExist(int slideId)
+        {
+            Slides slides = Globals.ThisAddIn.Application.ActivePresentation.Slides;
+            foreach (Slide slide in slides)
```

Comment:
For a future version, you could consider optimizing this, rather than checking every slide in the presentation, how about responding to the event of deleting a slide and checking if deleted slide is the elearninglab slide? I'm not super sure either but seems like a good place to optimise later.

Reply:
Yes you are right about this. I also have the same concern. Probably this function needs to be removed in a future version. However, I could not seem to find a `slide deleted event` handler for powerpoint, therefore, I tried to iterate through all slides for now.  

Another possible alternative is to add a try-catch block when requesting for `slide.ID` property. However, an exception will still be thrown from visual studio during run time if we are requesting for invalid instance from powerpoint object model. A try-catch block will work fine if we open PPT  directly instead of running from visual studio.